File: 6/main.java
class Solution {
    public List<Integer> valueEqualToIndex(List<Integer> nums) {
        List<Integer> result = new ArrayList<>();
        for (int i = 0; i < nums.size(); i++) {
            if (nums.get(i) == i + 1) { // Adjust index to 1-based
                result.add(i + 1);      // Add to result (adjust back to 1-based)
            }
        }
        return result;
    }
}

File: 6/main.py
class Solution:
    # Function to find values in array equal to their indices
    def valueEqualToIndex(self, arr):
        ans = []  # create an empty list to store the values
        n = len(arr)
        # iterate through the array
        for i in range(n):
            # check if the value at index i is equal to its index + 1
            if arr[i] == i + 1:
                ans.append(arr[i])  # if true, add the value to the list
        return ans  # return the list of values that are equal to their indices
    
    

